src/spi_comms.c

```diff
@@ -0,0 +1,364 @@
+#include "spi_comms.h"
+
+// SPI data from PAY-SSM
+uint8_t spi_first_byte = 0;
+uint8_t spi_second_byte = 0;
+
+uint8_t spi_spdr_tx_buffer = 0;    // holds data to be shifted into SPDR, for SPI transfer
+uint8_t spi_frame_number = 0;      // byte number of data trasmission
+
+uint8_t spi_status_byte = 0;
+
+
+// Initialize SPI comms as SPI slave, with interrupts enabled
+void init_spi_comms(void){
+    // initialize SPI
+    init_spi();
+
+    // set DATA_RDYn pin as output high
+    DATA_RDYn_DDR |= _BV(DATA_RDYn);    // set direction = output
+    opt_set_data_rdy_high();
+}
+
+
+// return status of SS pin
+// ------- might not need, since OPTICAL just checks SPIF flag
+uint8_t opt_check_ss_pin(void){
+    return SS_PIN >> SS; 
+}
+
+
+// set DATA_RDYn low
+void opt_set_data_rdy_low(){
+    DATA_RDYn_PORT &= ~_BV(DATA_RDYn);     // set output LOW
+}
+
+// set DATA_RDYn high
+void opt_set_data_rdy_high(){
+    DATA_RDYn_PORT |= _BV(DATA_RDYn);  // set output HIGH
+}
+
+
+// to be put in infinite loop in main
+void opt_loop(void){
+    // if SPI transfer if completed
+    if (SPSR & _BV(SPIF)){
+        spi_first_byte = SPDR;
+        opt_set_data_rdy_low();         // set DATA_RDYn low to signal to PAY that it's ready for the second byte
+
+        // wait until another SPI transfer is completed
+        // --> aka wait until SPIF is no longer high 
+        uint16_t timeout = UINT16_MAX;
+        while (!(SPSR & _BV(SPIF)) && timeout>0){
+            timeout--;
+        }
+
+        opt_set_data_rdy_high();        // clear DATA_RDYn, so PAY doesn't freak out
+        spi_second_byte = SPDR;
+
+        // now, got both bytes
+        // perform the requested command and send back data if necessary
+        manage_cmd(spi_first_byte);
+    }
+    opt_set_data_rdy_high();
+}
+
+
+// depending on cmd_code, does appropriate requested function + return data (if needed)
+void manage_cmd (uint8_t cmd_code){

```

Comment:
Add a second argument with the well number, which will only be used for the get reading command